component-catalog-connectors/mlx-connector/README.md

```diff
@@ -0,0 +1,63 @@
+## Machine Learning Exchange catalog connector
+
+This catalog connector enables Elyra to load Kubeflow Pipelines components from [Machine Learning Exchange](https://github.com/machine-learning-exchange) (MLX) deployments.
+
+### Install the connector
+
+You can install the MLX catalog connector from PyPI or source code. Note that a **rebuild of JupyterLab is not required**.
+
+**Prerequisites**
+
+- [Install Elyra](https://elyra.readthedocs.io/en/stable/getting_started/installation.html) (version 3.3 and above).
+- [Machine Learning Exchange deployment](https://github.com/machine-learning-exchange/mlx) ([quickstart guide](https://github.com/machine-learning-exchange/mlx/tree/main/quickstart))
+
+**Install from PyPI**
+
+  ```
+  $ pip install mlx-component-catalog-connector
+  ```
+
+**Install from source code**
+
+   ```
+   $ git clone https://github.com/elyra-ai/examples.git
+   $ cd examples/component-catalog-connectors/mlx-connector/
+   $ make source-install
+   ```
+
+### Use the connector
+
+1. Launch JupyterLab.
+1. [Open the `Manage Components` panel](
+https://elyra.readthedocs.io/en/stable/user_guide/pipeline-components.html#managing-custom-components-using-the-jupyterlab-ui).
+1. Add a new MLX component catalog ('`+`' > '`New Machine Learning Exchange Component Catalog`').
+1. Specify a catalog name, e.g. '`MLX dev catalog`'.
+1. Select Kubeflow Pipelines as runtime.
+1. (Optional) Specify a category under which the catalog's component will be organized in the palette.
+1. Configure the `MLX API URL`, e.g. '`http://my-mlx-server.mydomain:8080/`'. 
+   > Note: The Machine Learning Exchange API URL is different from the GUI URL!
+1. Apply an optional filter expression to the component names. The `*` (zero or more characters) and `?` (zero or one character) wildcards are supported.
+1. Save the catalog entry.
+1. Open the Kubeflow Pipelines Visual Pipeline Editor and expand the palette. The components that were loaded from the specified MLX URL are displayed.
+
+### Uninstall the connector
+
+1. Remove all MLX catalog entries from the '`Manage Components`' panel.
+1. Stop JupyterLab.
+1. Uninstall the `mlx-component-catalog-connector` package.
+   ```
+   $ pip uninstall -y mlx-component-catalog-connector
+   ```
+
+### Troubleshooting
+
+**Problem: The palette does not diplay any components from the configured catalog.**
+
+**Solution:** If the the Elyra GUI does not display any error message indicating that a problem was encountered, inspect the JupyterLab log file.
+
+Example error message (The specified MLX URL is invalid):
+
+```
+Error fetching component list from MLX catalog http://localhost:8080: ... Failed to establish a new connection: [Errno 61] Connection refused'
+```
+
```

Comment:
Thoughts on adding something here about this error that pops up in the GUI?

<img width="450" alt="Screen Shot 2021-10-29 at 1 16 25 PM" src="https://user-images.githubusercontent.com/31816267/139483248-0d17fde2-cdfc-4ebe-9b47-0e688daf0f79.png">

Since it doesn't include the traceback like many other errors that surface in the GUI, I feel like it might be sort of cryptic.

Reply:
I agree. I've been running into several places where tracebacks aren't even logged and it's difficult to troubleshoot.

Do you know what caused this particular error?

Reply:
I spoofed this one by not implementing one of the abstract methods in a CCC. But since all of that is first called during processor initialization, that's where the error surfaces. 

But yeah, at that point you sort of have no choice but to comb through the logs, or, if you happen to know to search for "Error registering processor... " that works too. Neither or which is very helpful

Reply:
The processor should not die if a connector implementation goes rogue. Log and continue (:

Reply:
Sounds good! Added to my list for Monday

Reply:
Marking this as resolved. The original intention was to provide catalog type specific troubleshooting help